app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/SingleDisposalsTriageController.scala

```diff
@@ -1599,59 +1638,60 @@ class SingleDisposalsTriageController @Inject() (
-                           ) EitherT.pure(())
-                           else
-                             returnsService.storeDraftReturn(
-                               newFillingOutReturn.draftReturn,
-                               newFillingOutReturn.subscribedDetails.cgtReference,
-                               newFillingOutReturn.agentReferenceNumber
-                             )
-                       )
-                  _ <- EitherT(
-                         updateSession(sessionStore, request)(
-                           _.copy(journeyStatus = Some(updatedState.merge))
-                         )
+    requiredField(triageAnswers) match {
+      case None    => Redirect(redirectToIfNotValidJourney(triageAnswers))
+      case Some(r) =>
+        form(r)
+          .bindFromRequest()
+          .fold(
+            formWithErrors =>
+              BadRequest(
+                page(state, triageAnswers, formWithErrors, state.isRight, r)
+              ),
+            { value =>
+              val updatedState = updateState(value, state, triageAnswers)
+
+              val result = for {
+                _ <- updatedState.fold(
+                       _ => EitherT.pure(()),
+                       newFillingOutReturn =>
+                         if (
+                           state.exists(
+                             _._2.draftReturn === newFillingOutReturn.draftReturn
+                           )
+                         ) EitherT.pure(())
+                         else
+                           returnsService.storeDraftReturn(
+                             newFillingOutReturn.draftReturn,
+                             newFillingOutReturn.subscribedDetails.cgtReference,
+                             newFillingOutReturn.agentReferenceNumber
+                           )
+                     )
+                _ <- EitherT(
+                       updateSession(sessionStore, request)(
+                         _.copy(journeyStatus = Some(updatedState.merge))
                        )
-                } yield ()
-
-                result.fold(
-                  { e =>
-                    logger.warn("Could not update session", e)
-                    errorHandler.errorResult()
-                  },
-                  _ =>
-                    Redirect(
-                      routes.SingleDisposalsTriageController.checkYourAnswers()
-                    )
-                )
-              }
-            )
-      }
+                     )
+              } yield ()
+
+              result.fold(
+                { e =>
+                  logger.warn("Could not update session", e)
+                  errorHandler.errorResult()
+                },
+                _ =>
+                  Redirect(
+                    routes.SingleDisposalsTriageController.checkYourAnswers()
+                  )
+              )
+            }
+          )
     }
 
   private def displayTriagePage[R, A, Page : Writeable](
```

Comment:
it would be good to have a comment to describe what these types `R` and `A` represent

Reply:
added some documentation now